sdk/monitor/query/azmetrics/utils_test.go

```diff
@@ -0,0 +1,149 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azmetrics_test
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/recording"
+	"github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azmetrics"
+	"github.com/stretchr/testify/require"
+)
+
+const recordingDirectory = "sdk/monitor/query/azmetrics/testdata"
+const fakeResourceURI = "/subscriptions/faa080af-c1d8-40ad-9cce-e1a451va7b87/resourceGroups/rg-example/providers/Microsoft.AppConfiguration/configurationStores/example"
+const fakeSubscrtiptionID = "faa080af-c1d8-40ad-9cce-e1a451va7b87"
+const fakeRegion = "westus"
+
+var (
+	credential     azcore.TokenCredential
+	resourceURI    string
+	subscriptionID string
+	region         string
+)
+
+func TestMain(m *testing.M) {
+	code := run(m)
+	os.Exit(code)
+}
+
+func run(m *testing.M) int {
+	if recording.GetRecordMode() == recording.PlaybackMode || recording.GetRecordMode() == recording.RecordingMode {
+		proxy, err := recording.StartTestProxy(recordingDirectory, nil)
+		if err != nil {
+			panic(err)
+		}
+		defer func() {
+			err := recording.StopTestProxy(proxy)
+			if err != nil {
+				panic(err)
+			}
+		}()
+	}
+
+	if recording.GetRecordMode() == recording.PlaybackMode {
+		credential = &FakeCredential{}
+	} else {
+		tenantID := getEnvVar("AZMETRICS_TENANT_ID", "")
+		clientID := getEnvVar("AZMETRICS_CLIENT_ID", "")
+		secret := getEnvVar("AZMETRICS_CLIENT_SECRET", "")
+		var err error
+		credential, err = azidentity.NewClientSecretCredential(tenantID, clientID, secret, nil)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	resourceURI = getEnvVar("RESOURCE_URI", fakeResourceURI)
+	subscriptionID = getEnvVar("AZMETRICS_SUBSCRIPTION_ID", fakeSubscrtiptionID)
+	region = getEnvVar("AZMETRICS_LOCATION", fakeRegion)
+
+	return m.Run()
+}
+
+func startRecording(t *testing.T) {
+	err := recording.Start(t, recordingDirectory, nil)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := recording.Stop(t, nil)
+		require.NoError(t, err)
+	})
+}
+
+func startTest(t *testing.T) *azmetrics.Client {
+	startRecording(t)
+	transport, err := recording.NewRecordingHTTPClient(t, nil)
+	require.NoError(t, err)
+	opts := &azmetrics.ClientOptions{ClientOptions: azcore.ClientOptions{Transport: transport}}
+	endpoint := "https://" + region + ".metrics.monitor.azure.com"
+	client, err := azmetrics.NewClient(endpoint, credential, opts)
+	if err != nil {
+		panic(err)
+	}
+	return client
+}
+
+func getEnvVar(envVar string, fakeValue string) string {
```

Comment:
Would [recording.GetEnvVar()](https://github.com/Azure/azure-sdk-for-go/blob/b08f175d31a9a7d48edbb8f76ce1a1721f2e4232/sdk/internal/recording/recording.go#L131) work?

Reply:
Would need to look into. Maybe in a separate PR?